com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/HooksPhaseBuilder.java

```diff
@@ -0,0 +1,65 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import com.sap.cloud.lm.sl.cf.core.model.HookPhase;
+import com.sap.cloud.lm.sl.cf.core.model.HookPhaseProcessType;
+import com.sap.cloud.lm.sl.cf.core.model.HookProcessPhase;
+import com.sap.cloud.lm.sl.cf.core.model.Phase;
+import com.sap.cloud.lm.sl.cf.core.model.SubprocessPhase;
+import com.sap.cloud.lm.sl.cf.process.steps.ProcessContext;
+import com.sap.cloud.lm.sl.cf.process.variables.Variables;
+import com.sap.cloud.lm.sl.cf.web.api.model.ProcessType;
+
+@Named
+public class HooksPhaseBuilder {
+
+    private static final String HOOKS_DELIMITER = ".";
+    private static final String DEFAULT_HOOK_ENTITY = "application";
+    private final ProcessTypeParser processTypeParser;
+
+    @Inject
+    public HooksPhaseBuilder(ProcessTypeParser processTypeParser) {
+        this.processTypeParser = processTypeParser;
+    }
+
+    public List<HookPhase> buildHookPhases(List<HookPhase> hookPhases, ProcessContext context) {
+        return hookPhases.stream()
+                         .map(hookPhase -> buildPhase(hookPhase, context))
+                         .map(HookPhase::fromString)
+                         .collect(Collectors.toList());
+    }
+
+    private String buildPhase(HookPhase hookPhase, ProcessContext context) {
```

Comment:
wouldn't it make a bit more sense to have this code in the enum for the HookPhase itself?

Reply:
for example: HOOKS_DELIMITER is something already used in the enum, as well as the enum has a method to get a value from 1 string, why not add a method to get a value from all the needed strings (deploymentType, phase, phaseLocator)